File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        res = []
        c = 0
        for i in reversed(range(len(digits))):
            cur = digits[i]
            if i == len(digits) - 1:
                cur += 1
            cur += c
            res.append(cur%10)
            c = cur // 10
        if c:
            res.append(c)
        return res[::-1]
            
            
            
        

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {

        int n = digits.length;

        int car = 0;

        List<Integer> res = new ArrayList<>();

        for (int i = n-1; i >=0 ; i--) {
            int dig = digits[i];
            if (i == n-1) {
                dig += 1;
            }
            dig += car;
            int curr = dig % 10;
            car = (int) dig/10;

            res.add(curr);
        }

        if (car != 0) {
            res.add(car);
        }

        Collections.reverse(res);

        return res.stream().mapToInt(i -> i).toArray();
        
    }
}